### ISSUE TITLE
Tooltip still leaking memory (or DOM nodes)

### ISSUE THREAD

**Issue by afinne** [twbs/bootstrap#16451]
The Tooltip still seems to leak a bit, even after the latest PRs that set ```$viewport``` to null. What we did in our project was to also clear out the $element. This, however required also an extra check in the complete function of the hide function:

    function complete() {
      if (that.hoverState != 'in') $tip.detach()
      if (that.$element) {
        that.$element
          .removeAttr('aria-describedby')
          .trigger('hidden.bs.' + that.type)
      }
      callback && callback()
    }

And to be extra sure that we are not calling destroy too many times:

    Tooltip.prototype.destroy = function () {
      var that = this
      clearTimeout(this.timeout)
      if (this.isDestroying == true) {
        return
      }
      this.isDestroying = true
      this.hide(function () {
        that.$element.off('.' + that.type).removeData('bs.' + that.type)
        if (that.$tip) {
          that.$tip.detach()
        }
        that.$tip = null
        that.$arrow = null
        that.$viewport = null
        that.$element = null
      })

**Comment by zacechola** [comment]
X-Ref: #16434

**Comment by zacechola** [comment]
Like the Tab post cross-referenced above, we need to see an example of this leak in a testable environment like JS Bin or similar. I can't recreate the leaks using vanilla Bootstrap. I'm guessing I'm missing something that you're using in your environment that could be contributing.

**Comment by fat** [comment]
yep please reopen an issue with a workable example. 

cheers

**Comment by animafact** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

animafact@naver.com 님께 보내신 메일 <Re: [bootstrap] Tooltip still leaking memory (or DOM nodes) (#16451)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------
